### PR TITLE
[grpc-php] php8 compatability

### PR DESCRIPTION
**Pull request by Alexander1000**
fix compatability with php8

@yashykt

### CODE REVIEW

**Comment by stanley-cheung** [comment]
Hi @Alexander1000 , thanks for the contributions. But is there already a PHP 8 development version out that we can try compiling this against?

**Comment by Alexander1000** [comment]
Hi, @stanley-cheung 
I compiled with master branch of php.
It is already positioned as an early version 8.
In doc say about this changes for php8:
https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS
`php -v
PHP 8.0.0-dev (cli) (built: Oct 22 2019 09:34:37) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.0-dev, Copyright (c) Zend Technologies
`

**Comment by stanley-cheung** [comment]
@Alexander1000 I see. Let me check that out.

Is there anything else we need to do for PHP 8? There seems to be quite a list in the "upgrading internals" doc you linked. Or is this PR the minimal to bring the extension to at least compiling?

**Comment by Alexander1000** [comment]
Yes. This is enough for this extension to compile and work.

**Comment by Alexander1000** [comment]
If you want to use the composer, then most likely even openssl.
I  build for home projects:
`
./configure
  --with-openssl
  --enable-pcntl
  --enable-bcmath
  --with-pcre-jit
  --with-curl
  --enable-soap
  --with-pdo-pgsql
`

File: src/php/ext/grpc/completion_queue.h [hunk 1]
```diff
@@ -23,6 +23,10 @@
 
 #include <grpc/grpc.h>
 
+#if PHP_MAJOR_VERSION >= 8
```
[the remark this answers is not in the document]
Comment (Alexander1000) [reply src/php/ext/grpc/completion_queue.h]: The rest of the files include php_grpc.h in which there is a defined of the macros used.

**Comment by Alexander1000** [comment]
Hello! PHP 8 in this time already have release candidates.

**Comment by janysp** [comment]
Would be nice to have the GRPC  module in our PHP 8.0.0 RC2 tests

**Pull request closed without merging**